src/main/java/net/minecraftforge/client/model/renderable/SimpleRenderable.java

```diff
@@ -0,0 +1,161 @@
+ * Each mesh pairs a texture, with a set of quads.
+ */
+public class SimpleRenderable implements IRenderable<MultipartTransforms>
+{
+    private final List<Part> parts = new ArrayList<>();
+
+    private SimpleRenderable()
+    {
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    @Override
+    public void render(PoseStack poseStack, MultiBufferSource bufferSource, Function<ResourceLocation, RenderType> renderTypeFunction, int lightmapCoord, int overlayCoord, float partialTicks, MultipartTransforms renderValues)
+    {
+        for(var part : parts)
+        {
+            part.render(poseStack, bufferSource, renderTypeFunction, lightmapCoord, overlayCoord, renderValues);
+        }
+    }
+
+    private static class Part
+    {
+        private final String name;
+        private final List<Part> parts = new ArrayList<>();
+        private final List<Mesh> meshes = new ArrayList<>();
+
+        public Part(String name)
+        {
+            this.name = name;
+        }
+
+        public void render(PoseStack poseStack, MultiBufferSource bufferSource, Function<ResourceLocation, RenderType> renderTypeFunction, int lightmapCoord, int overlayCoord, MultipartTransforms renderValues)
+        {
+            Matrix4f matrix = renderValues.getPartValues(name);
+            if (matrix != null)
+            {
+                poseStack.pushPose();
+                poseStack.mulPoseMatrix(matrix);
+            }
+
+            for(var part : parts)
+            {
+                part.render(poseStack, bufferSource, renderTypeFunction, lightmapCoord, overlayCoord, renderValues);
+            }
+
+            for(var mesh : meshes)
+            {
+                mesh.render(poseStack, bufferSource, renderTypeFunction, lightmapCoord, overlayCoord, renderValues);
+            }
+
+            if (matrix != null)
+            {
+                poseStack.popPose();
+            }
+        }
+    }
+
+    private static class Mesh
+    {
+        private final ResourceLocation texture;
+        private final List<BakedQuad> quads = new ArrayList<>();
+
+        public Mesh(ResourceLocation texture)
+        {
+            this.texture = texture;
+        }
+
+        public void render(PoseStack poseStack, MultiBufferSource bufferSource, Function<ResourceLocation, RenderType> renderTypeFunction, int lightmapCoord, int overlayCoord, MultipartTransforms renderValues)
+        {
+            var consumer = bufferSource.getBuffer(renderTypeFunction.apply(texture));
+            for(var quad : quads)
+            {
+                consumer.putBulkData(poseStack.last(), quad, 1, 1, 1, 1, lightmapCoord, overlayCoord, true);
+            }
+        }
+    }
+
+    public static class Builder
+    {
+        private final SimpleRenderable renderable = new SimpleRenderable();
+
+        private Builder()
+        {
+        }
+
+        public PartBuilder child(String name)
```

Comment:
I don't see any issues with this. Though the chainability of the builder is lost if there is no reference back to the builder from the internal part.

Reply:
I was hoping to avoid having an ungodly recursive generic but I guess it's not as bad as it looked in my head.
It does however mean that if you are a few levels deep, the actual type of PartBuilder will be `PartBuilder<PartBuilder<PartBuilder<...<Builder>...>>>` but 🤷‍♂️ you don't have to look at the type.